### PR TITLE
Group entries filters

### PR DESCRIPTION
**Pull request by AdityaKhatri**
* Group entry filters
* Change empty notification message

### CODE REVIEW

File: src/redux/reducers/siloDomainData/entries.js [hunk 1]
```diff
@@ -245,6 +252,34 @@ const entryViewSetFilter = (state, action) => {
...
+            safeState.entriesView[projectId] = {};
+        }
+        if (!safeState.entriesView[projectId].filter) {
+            // eslint-disable-next-line no-param-reassign
+            safeState.entriesView[projectId].filter = {};
+        }
+        // eslint-disable-next-line no-param-reassign
+        safeState.entriesView[projectId].filter = {
+            ...safeState.entriesView[projectId].filter,
+            ...{ [filterKey]: newValue },
```
Comment (tnagorra) [inline src/redux/reducers/siloDomainData/entries.js]:
```suggestion
            [filterKey]: newValue,
```

File: src/views/Entries/FilterEntriesForm/FrameworkFilter/index.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    if (!filter || !filter.type) {
+        return null;
+    }
+
+    const propsForFilter = {
+        key: filterKey,
+        className: styles.entriesFilter,
+        label: title,
+        onChange: values => handleFilterChange(filterKey, values),
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/FrameworkFilter/index.js]: Why not use useCallback?

File: src/views/Entries/FilterEntriesForm/FrameworkFilter/index.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                    regions={regions}
+                />
+            );
+        }
+        case 'multiselect':
+            return (
+                <MultiSelectInput
+                    {...propsForFilter}
+                    value={propsForFilter.value || emptyList}
+                    options={filter.options || emptyList}
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/FrameworkFilter/index.js]: Do we need this?

Comment (tnagorra) [reply src/views/Entries/FilterEntriesForm/FrameworkFilter/index.js]: By this I mean setting default values.
Also, why not use `??` syntax? I think it is already available in `develop`.

File: src/views/Entries/FilterEntriesForm/index.js [hunk 4]
```diff
@@ -129,370 +129,380 @@ const requestOptions = {
...
-        const { pristine } = this.state;
-
-        if (pristine) {
-            const { unsetEntriesViewFilter } = this.props;
+const filterKeySelector = (d = {}) => d.key;
+
+const optionLabelSelector = (d = {}) => d.value;
+const optionKeySelector = (d = {}) => d.key;
+const optionTitleSelector = (d = {}) => d.title;
+const optionIdSelector = (d = {}) => d.id;
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/index.js]: We should use optional chaining here!

File: src/views/Entries/FilterEntriesForm/index.js [hunk 5]
```diff
@@ -129,370 +129,380 @@ const requestOptions = {
...
+    }, [setShowFilters]);
+
+    const appliedFiltersLabel = useMemo(() => {
+        const appliedFiltersKeys = mapToList(
+            entriesFilters,
+            (d, k) => (doesObjectHaveNoData(d, ['']) ? undefined : k),
+        ).filter(isDefined);
+        const frameworkFiltersTitle = listToMap(filtersFromProps, d => d.key, d => d.title);
+        const staticFilters = appliedFiltersKeys
+            .map(f => staticFiltersLabelMap[f] || frameworkFiltersTitle[f])
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/index.js]:
```suggestion
            .map(f => staticFiltersLabelMap[f] ?? frameworkFiltersTitle[f])
```

File: src/views/Entries/FilterEntriesForm/index.js [hunk 6]
```diff
@@ -129,370 +129,380 @@ const requestOptions = {
...
+            .filter(isDefined);
+
+        return ([
+            ...staticFilters,
+        ]);
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/index.js]:
```suggestion
        return staticFilters;
```

File: src/views/Entries/FilterEntriesForm/index.js [hunk 7]
```diff
@@ -129,370 +129,380 @@ const requestOptions = {
...
+                            <>
+                                <MultiSelectInput
+                                    className={styles.entriesFilter}
+                                    keySelector={optionIdSelector}
+                                    labelSelector={optionTitleSelector}
+                                    options={projectEntryLabel}
+                                    label={staticFiltersLabelMap.project_entry_labels}
+                                    onChange={(value) => { handleFilterChange('project_entry_labels', value); }}
+                                    showHintAndError={false}
+                                    value={filters.project_entry_labels || emptyList}
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/index.js]: Prefer `??` to set default values

File: src/views/Entries/FilterEntriesForm/index.js [hunk 8]
```diff
@@ -129,370 +129,380 @@ const requestOptions = {
...
+                        />
+                        <MultiSelectInput
+                            className={styles.entriesFilter}
+                            keySelector={optionKeySelector}
+                            labelSelector={optionLabelSelector}
+                            options={entryTypeOptions}
+                            label={staticFiltersLabelMap.entry_type}
+                            onChange={(value) => { handleFilterChange('entry_type', value); }}
+                            showHintAndError={false}
+                            value={filters.entry_type || emptyList}
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/index.js]: Prefer `??` to set default values

File: src/views/Entries/FilterEntriesForm/index.js [hunk 9]
```diff
@@ -129,370 +129,380 @@ const requestOptions = {
...
-                    </>
+                        <SelectInput
+                            className={styles.entriesFilter}
+                            keySelector={optionKeySelector}
+                            labelSelector={optionLabelSelector}
+                            options={commentStatusOptions}
+                            label={staticFiltersLabelMap.comment_status}
+                            onChange={(value) => { handleFilterChange('comment_status', value); }}
+                            showHintAndError={false}
+                            value={filters.comment_status || undefined}
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/index.js]: I do not understand this behavior

File: src/views/Entries/FilterEntriesForm/index.js [hunk 10]
```diff
@@ -129,370 +129,380 @@ const requestOptions = {
...
                             className={styles.entriesFilter}
-                            label={_ts('entries', 'entryGroupsFilterLabel')}
-                            onChange={(value) => { this.handleFilterChange('lead_group_label', value); }}
+                            keySelector={optionKeySelector}
+                            labelSelector={optionLabelSelector}
+                            options={createdBy}
+                            label={staticFiltersLabelMap.comment_created_by}
+                            onChange={(value) => { handleFilterChange('comment_created_by', value); }}
                             showHintAndError={false}
+                            value={filters.comment_created_by || emptyList}
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/index.js]: Prefer ?? to set default values

File: src/views/Entries/FilterEntriesForm/index.js [hunk 11]
```diff
@@ -129,370 +129,380 @@ const requestOptions = {
...
+        if (applyOnChange) {
+            updateEntriesViewFilter({
+                filterKey: key,
+                newValue: values,
+            });
+        } else {
+            setPristine(false);
+            setFilters(oldFilters => ({
+                ...oldFilters,
+                ...{ [key]: values },
```
Comment (tnagorra) [inline src/views/Entries/FilterEntriesForm/index.js]:
```suggestion
                ...[key]: values,
```

**Review by samshara — APPROVED** [review]

**Pull request merged**